### ISSUE TITLE
SysTraceWorker accumulates samples even though TRACY_ON_DEMAND is set

### ISSUE THREAD

**Issue by tlubin** [wolfpld/tracy#318]
When TRACY_ON_DEMAND is set, I would expect no sampling to occur until a profiler is connected. The data collected in ScopedZone clearly handles TRACY_ON_DEMAND properly:

```
    tracy_force_inline ScopedZone( const SourceLocationData* srcloc, bool is_active = true )
#ifdef TRACY_ON_DEMAND
        : m_active( is_active && GetProfiler().IsConnected() )
#else
        : m_active( is_active )
#endif
    {
        if( !m_active ) return;
```
However, the SysTraceWorker reads perf events and saves them into the thread's ConcurrentQueue with no regard for TRACY_ON_DEMAND. This results in unbounded memory growth until you connect a profiler.

**Comment by wolfpld** [comment]
Should be fixed in f058ad01.

**Comment by tlubin** [comment]
Confirmed, thanks for the quick fix!
